custom/icds_reports/tasks.py

```diff
@@ -967,6 +973,90 @@ def collect_inactive_awws():
     celery_task_logger.info("Ended updating the Inactive AWW")
 
 
+@periodic_task(run_every=crontab(day_of_week=6, hour=18, minute=30), acks_late=True, queue='icds_aggregation_queue')
+def collect_inactive_dashboard_users():
+    celery_task_logger.info("Started updating the Inactive Dashboard users")
+
+    end_date = datetime.utcnow()
+    start_date_week = end_date - timedelta(days=7)
+    start_date_month = end_date - timedelta(days=30)
+
+    not_logged_in_week = get_dashboard_users_not_logged_in(start_date_week, end_date)
+    not_logged_in_month = get_dashboard_users_not_logged_in(start_date_month, end_date)
+
+    week_file_name = 'dashboard_users_not_logged_in_{:%Y-%m-%d}_to_{:%Y-%m-%d}.csv'.format(
+        start_date_week, end_date
+    )
+    month_file_name = 'dashboard_users_not_logged_in_{:%Y-%m-%d}_to_{:%Y-%m-%d}.csv'.format(
+        start_date_month, end_date
+    )
+    rows_not_logged_in_week = _get_inactive_dashboard_user_rows(not_logged_in_week)
+    rows_not_logged_in_month = _get_inactive_dashboard_user_rows(not_logged_in_month)
+
+    sync = IcdsFile(blob_id="inactive_dashboad_users_%s.zip" % date.today().strftime('%Y-%m-%d'),
+                    data_type='inactive_dashboard_users')
+
+    in_memory = BytesIO()
+    zip_file = zipfile.ZipFile(in_memory, 'w', zipfile.ZIP_DEFLATED)
+
+    zip_file.writestr(week_file_name,
+                      '\n'.join(rows_not_logged_in_week)
+                      )
+    zip_file.writestr(month_file_name,
+                      '\n'.join(rows_not_logged_in_month)
+                      )
+
+    zip_file.close()
+
+    # we need to reset buffer position to the beginning after creating zip, if not read() will return empty string
+    # we read this to save file in blobdb
+    in_memory.seek(0)
+    sync.store_file_in_blobdb(in_memory)
+
+    sync.save()
+
+
+def _get_inactive_dashboard_user_rows(not_logged_in_week):
+    from corehq.apps.users.models import CommCareUser
+    rows = ['"Username", "Location", "State"']
+    for username in not_logged_in_week:
+        user = CommCareUser.get_by_username(username)
+        loc = user.sql_location
+        loc_name = loc.name if loc else ''
+        state = loc.get_ancestor_of_type('state') if loc else None
+        state_name = state.name if state else ''
+        rows.append('"{}", "{}", "{}"'.format(username, loc_name, state_name))
+
+    return rows
+
+
+def get_dashboard_users_not_logged_in(start_date, end_date, domain='icds-cas'):
+
+    all_users = get_all_user_id_username_pairs_by_domain(domain, include_web_users=False,
+                                                         include_mobile_users=True)
+
+    dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
```

Comment:
feels like there should be a better way to do this - can we check with GS if there isn't a field in custom user data or a role that we can use to filter users better? Then we could use ES instead of having to iterate through ALL usernames.

Reply:
Hey I talked to Kishan about this. He said If a mobile user is assigned any role. He has the access to dashboard which we can use as a filter. But, I think we should stick to this. Because Using the other filter there are other users also which are test users. which I m not sure how Will we filter out.
@snopoke  does it sound good.